pennylane_lightning/src/tests/TestHelpers.hpp

```diff
@@ -62,3 +68,45 @@ void scaleVector(std::vector<std::complex<Data_t>> &data,
         data.begin(), data.end(), data.begin(),
         [scalar](const std::complex<Data_t> &c) { return c * scalar; });
 }
+
+/**
+ * @brief create |0>^N
+ */
+template <typename fp_t>
+auto create_zero_state(size_t num_qubits) -> std::vector<std::complex<fp_t>> {
+    std::vector<std::complex<fp_t>> res(1U << num_qubits, 0.0);
+    res[0] = std::complex<fp_t>{1.0, 0.0};
+    return res;
+}
+
+/**
+ * @brief create |+>^N
+ */
+template <typename fp_t>
+auto create_plus_state(size_t num_qubits) -> std::vector<std::complex<fp_t>> {
+    std::vector<std::complex<fp_t>> res(1U << num_qubits, 1.0);
+    for (auto &elt : res) {
+        elt /= std::sqrt(1U << num_qubits);
+    }
+    return res;
+}
+
+/**
+ * @brief create a random state
+ */
+template <typename fp_t, class RandomEngine>
+auto create_random_state(RandomEngine &re, size_t num_qubits)
+    -> std::vector<std::complex<fp_t>> {
+    std::vector<std::complex<fp_t>> res(1U << num_qubits, 0.0);
+    std::normal_distribution<fp_t> dist;
```

Comment:
Should we opt for a uniform distribution here, rather than normal? And maybe using a Mersenne-Twister?

Reply:
I used a normal distribution as it generates a Haar random state. But of course, as this is just for the test purpose, uniform sounds better. The random engine is passed from the caller. It now uses `std::default_random_engine` (which is linear congruential generator in GCC) but I can change it to Mersenne-Twister.

Reply:
Yea, I reckon that would be a good idea to use MT. I think for large states, and for better distributed randomness the MT algorithm is a better call with both statistical tests and overall runtime performance.

Reply:
I change `std::default_random_engine` to `std::mt19937` in the test codes.